website/project/model.py

```diff
@@ -3039,3 +3034,137 @@ def approve_embargo(self, user, token):
+    #    def is_approved(self, value):
+    #        self.flags['isApproved'] = value
+
+    def update_metadata(self, metadata, save=True):
+        # TODO: uncommnet to disallow editing drafts that are approved or pending approval
+        # if self.is_pending_review or self.is_approved:
+        #    raise HTTPError(http.BAD_REQUEST)
+        changes = []
+        for key, value in metadata.iteritems():
+            old_value = self.registration_metadata.get(key)
+            if not old_value or old_value.get('value') != value.get('value'):
+                changes.append(key)
+        self.registration_metadata.update(metadata)
+        if save:
+            self.save()
+        # TODO: uncomment to nullify approval state if edited
+        # self.after_edit(changes)
+
+    def before_edit(self, auth):
+        messages = []
+        # TODO: uncomment to provide pre-edit warnings for drafts that are approved or are pending approval
+        # if self.flags.get('isApproved'):
+        #     messages.append('The draft registration you are editing is currently approved. Please note that if you make any changes (excluding comments) this approval status will be revoked and you will need to submit for approval again.')
+        # if self.flags.get('isPendingReview'):
+        #     messages.append('The draft registration you are editing is currently pending review. Please note that if you make any changes (excluding comments) this request will be cancelled and you will need to submit for approval again.')
+        return messages
+
+    def after_edit(self, changes):
+        # revoke approval and review status if changed
+        if changes:
+            self.flags.update({
+                'isPendingReview': False,
+                'isApproved': False
+            })
+            self.save()
+
+    def find_question(self, qid):
+        for page in self.registration_schema.schema['pages']:
+            for question_id, question in page['questions'].iteritems():
+                if question_id == qid and 'description' in question:
+                    return question
+
+    def get_comments(self):
+        """ Returns a list of all comments made on a draft in the format of :
+        [{
+          [QUESTION_ID]: {
+            'question': [QUESTION],
+            'comments': [LIST_OF_COMMENTS]
+            }
+        },]
+        """
+
+        all_comments = []
+        for question_id, value in self.registration_metadata.iteritems():
+            all_comments.append({
+                question_id: {
+                    'question': self.find_question(question_id),
+                    'comments': value['comments'] if 'comments' in value else ''
+                }
+            })
+        return all_comments
+
+    def get_flat_comments(self):
+        """ Returns a flat list of all comments made on a draft
+        """
+        flat_comments = list()
+        for question_id, value in self.registration_metadata.iteritems():
```

Comment:
`question_id` doesn't seem to be used here. You can refactor as a list comprehension:

```python
return [comment for meta in self.registration_metadata.values() 
            for comment in meta.get('comments'), [])]
```